.github/workflows/ci.yml

```diff
@@ -0,0 +1,92 @@
+name: Run checks on pull requests and pushes on main branch
+
+on:
+  pull_request: {}
+  push:
+    branches: [main]
+
+jobs:
+  tests:
+    name: Checks of unit tests and format checks
```

Comment:
You might want to consider a shorter name here, for readability. If you look here, you can see that the names get truncated, and it's difficult to tell which run is which platform/Python version: https://github.com/godaddy/sample-size/runs/4230160107?check_suite_focus=true

One thing you might consider is something similar to what I did with `tartufo`, splitting the unit tests and linters into distinct jobs. That also allows them to fail separately, so that the contributor knows more easily whether a test or a linter failed. 

See here: https://github.com/godaddy/tartufo/blob/3b4acef809158676ebf7dfa8b93ae6f0a525b199/.github/workflows/ci.yml#L11
And here: https://github.com/godaddy/tartufo/blob/3b4acef809158676ebf7dfa8b93ae6f0a525b199/.github/workflows/ci.yml#L62